integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/support/ws/smartmetering/management/GetGsmDiagnosticResponseValidator.java

```diff
@@ -0,0 +1,101 @@
+      final String expectedValue = expectedEntyValue.getValue();
+      switch (expectedEntyValue.getKey()) {
+        case "operator":
+          assertThat(response.getOperator()).isEqualTo(expectedValue);
+          break;
+        case "modemRegistrationStatus":
+          assertThat(response.getModemRegistrationStatus())
+              .isEqualTo(ModemRegistrationStatus.fromValue(expectedValue));
+          break;
+        case "circuitSwitchedStatus":
+          assertThat(response.getCircuitSwitchedStatus())
+              .isEqualTo(CircuitSwitchedStatus.fromValue(expectedValue));
+          break;
+        case "packetSwitchedStatus":
+          assertThat(response.getPacketSwitchedStatus())
+              .isEqualTo(PacketSwitchedStatus.fromValue(expectedValue));
+          break;
+        case "cellId":
+          assertThat(response.getCellInfo().getCellId()).isEqualTo(Long.parseLong(expectedValue));
+          break;
+        case "locationId":
+          assertThat(response.getCellInfo().getLocationId())
+              .isEqualTo(Long.parseLong(expectedValue));
+          break;
+        case "signalQuality":
+          assertThat(response.getCellInfo().getSignalQuality())
+              .isEqualTo(SignalQuality.fromValue(expectedValue));
+          break;
+        case "bitErrorRate":
+          assertThat(response.getCellInfo().getBitErrorRate())
+              .isEqualTo(BitErrorRate.fromValue(expectedValue));
+          break;
+        case "mobileCountryCode":
+          assertThat(response.getCellInfo().getMobileCountryCode())
+              .isEqualTo(Long.parseLong(expectedValue));
+          break;
+        case "mobileNetworkCode":
+          assertThat(response.getCellInfo().getMobileNetworkCode())
+              .isEqualTo(Long.parseLong(expectedValue));
+          break;
+        case "channelNumber":
+          assertThat(response.getCellInfo().getChannelNumber())
+              .isEqualTo(Long.parseLong(expectedValue));
+          break;
+        case "numberOfAdjacentCells":
+          assertThat(response.getAdjacentCells().size()).isEqualTo(Long.parseLong(expectedValue));
+          break;
+        case "adjacentCellId":
+          assertThat(response.getAdjacentCells().get(0).getCellId())
+              .isEqualTo(Long.parseLong(expectedValue));
+          break;
+        case "adjacentCellSignalQuality":
+          assertThat(response.getAdjacentCells().get(0).getSignalQuality())
+              .isEqualTo(SignalQuality.fromValue(expectedValue));
+          break;
+        case "captureTime":
+          try {
+            assertThat(response.getCaptureTime())
+                .isEqualTo(DatatypeFactory.newInstance().newXMLGregorianCalendar(expectedValue));
+          } catch (final DatatypeConfigurationException e) {
+            fail("Error while validating capture time :", e);
+          }
+          break;
+        default:
+          fail("Unexpected value: " + expectedEntyValue.getKey());
```

Comment:
I don't fully get this. If a feature uses a value that is not checked here, then we want to have an error message, right? Otherwise we might accidentaly add a wrong value in the feature and don't know that is isn't checked. Or am I missing something?

Reply:
Validator is removed